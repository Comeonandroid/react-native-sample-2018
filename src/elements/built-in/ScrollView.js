import React from 'react'
import styled from 'styled-components'
import { PropTypes } from 'prop-types'


const ScrollView = ({ children, ...rest }) => (
  <Element {...rest}>
    {children}
  </Element>
)

ScrollView.propTypes = {
  children: PropTypes.node
}

const Element = styled.ScrollView `
`

export default ScrollView
