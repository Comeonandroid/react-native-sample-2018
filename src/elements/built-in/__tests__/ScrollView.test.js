import 'react-native'
import React from 'react'
import { itBehavesLike } from '../../../specs/shared_examples/index'

import ScrollView from '../ScrollView'

describe('<ScrollView />', () => {
  itBehavesLike('aThemedComponent', { subject: <ScrollView /> })
  itBehavesLike('aComponentWithChildren', { subject: <ScrollView /> })
})
