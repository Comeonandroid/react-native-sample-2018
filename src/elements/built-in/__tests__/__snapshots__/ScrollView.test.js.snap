// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScrollView /> dark: renders correctly 1`] = `
<RCTScrollView
  style={
    Array [
      Object {},
      undefined,
    ]
  }
>
  <View />
</RCTScrollView>
`;

exports[`<ScrollView /> light: renders correctly 1`] = `
<RCTScrollView
  style={
    Array [
      Object {},
      undefined,
    ]
  }
>
  <View />
</RCTScrollView>
`;

exports[`<ScrollView /> renders component when passed in 1`] = `
<Styled(ScrollView)>
  <Component />
</Styled(ScrollView)>
`;

exports[`<ScrollView /> renders text when passed in 1`] = `
<Styled(ScrollView)>
  Test
</Styled(ScrollView)>
`;
